dali/test/python/test_operator_normal_distribution.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+import numpy as np
+import scipy.stats as st
+
+
+class NormalDistributionPipeline(Pipeline):
+    def __init__(self, premade_batch, num_threads=1):
+        super(NormalDistributionPipeline, self).__init__(len(premade_batch), num_threads, 0)
+        self.premade_batch = premade_batch
+        self.ext_src = ops.ExternalSource()
+        self.norm = ops.NormalDistribution(device="cpu", dtype=types.FLOAT)
```

Comment:
How about testing externally provided shape and no shape at all?

Reply:
+1. And I'd also make the data type an input argument

Reply:
Added proper tests. As for making `dtype` per-sample, I don't know if it's really neccessary. Is there any imaginable use-case for that?